### PR TITLE
New search page

### PR DESCRIPTION
**Pull request by j8seangel**
TO DO:
- [ ] Connect with new dataset/endpoint
- [ ] Test auto-complete
- [ ] Don't request dataviews and datasets if already available
- [ ] Store advanced filters in URL to make the results shareable
- [x] First page load doesn't fetch resources
- [x] Send separated fields in Advanced mode
- [x] move search mode to url
- [x] move selectedVessels to redux

Screenshot 2023-04-14 at 10 26 43

### CODE REVIEW

File: apps/fishing-map/features/search/SearchAdvanced.tsx [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  fetchMoreResults,
+  setSearchQuery,
+  searchQuery,
+  debouncedQuery,
+  vesselsSelected,
+}: SearchComponentProps) {
+  const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+  const { searchPagination, searchSuggestion, searchSuggestionClicked } = useSearchConnect()
+  const basicSearchAllowed = useSelector(isBasicSearchAllowed)
```
Comment (j8seangel) [inline apps/fishing-map/features/search/SearchAdvanced.tsx:39]: Why is this needed here?

File: apps/fishing-map/features/search/SearchAdvanced.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+  const [searchState, setSearchState] = useState({
+    name: '',
+    mmsi: '',
+    imo: '',
+    callsign: '',
+    owner: '',
+  })
+
+  const onConfirm = (e: React.MouseEvent) => {
+    // setSearchQuery(e.target.value)
```
Comment (j8seangel) [inline apps/fishing-map/features/search/SearchAdvanced.tsx:53]: I'd remove this old code in case we are not going to use it anymore

File: apps/fishing-map/features/search/Search.tsx [hunk 3]
```diff
@@ -40,82 +23,59 @@ import {
...
   const heatmapDataviews = useSelector(selectActiveHeatmapDataviews)
-  const basicSearchAllowed = useSelector(isBasicSearchAllowed)
-  const advancedSearchAllowed = useSelector(isAdvancedSearchAllowed)
-  const searchResults = useSelector(selectSearchResults)
-  const searchStatus = useSelector(selectSearchStatus)
-  const searchStatusCode = useSelector(selectSearchStatusCode)
   const gfwUser = useSelector(isGFWUser)
-  const hasSearchFilters = checkSearchFiltersEnabled(searchFilters)
-  const vesselDataviews = useSelector(selectVesselsDataviews)
   const [vesselsSelected, setVesselsSelected] = useState<VesselWithDatasets[]>([])
```
Comment (j8seangel) [inline apps/fishing-map/features/search/Search.tsx:69]: As we need to drill down this state through Search => SearchResults, do you think worth it to move to a store reducer?

File: apps/fishing-map/features/search/search.slice.ts [hunk 4]
```diff
@@ -58,8 +59,12 @@ const initialState: SearchState = {
   data: null,
   suggestion: null,
   suggestionClicked: false,
+  option: 'basic',
```
Comment (j8seangel) [inline apps/fishing-map/features/search/search.slice.ts:62]: Why not store this in the URL?

File: apps/fishing-map/features/workspace/vessels/VesselsSection.tsx [hunk 5]
```diff
@@ -29,8 +31,12 @@ function VesselsSection(): React.ReactElement {
       category: TrackCategory.SearchVessel,
       action: 'Click search icon to open search panel',
     })
-    dispatchQueryParams({ query: '' })
-  }, [dispatchQueryParams])
+    dispatchLocation(SEARCH, {
```
Comment (j8seangel) [inline apps/fishing-map/features/workspace/vessels/VesselsSection.tsx:34]: ⚠️ we can't forget to update this part before merging

File: package.json [hunk 6]
```diff
@@ -33,10 +33,14 @@
     "prepare": "husky install"
   },
   "dependencies": {
+    "@emotion/react": "^11.10.6",
```
Comment (j8seangel) [inline package.json:36]: is this a `material-react-table` dependency ?

File: apps/fishing-map/features/search/search.slice.ts (hunk 4 above)
Comment (satellitestudiodesign) [reply apps/fishing-map/features/search/search.slice.ts:62]: good question, it should!

File: apps/fishing-map/features/search/Search.tsx (hunk 3 above)
Comment (satellitestudiodesign) [reply apps/fishing-map/features/search/Search.tsx:69]: makes sense

**Review by weberjavi — CHANGES_REQUESTED** [review]
I left some small comments related to the code.
What I've found out is that when scrolling to the bottom of the last page of a paginated search an error screen is displayed and the search results disappear from screen. Maybe it is related to the `fetchMoreOnBottomReached` function triggered on `muiTableContainerProps` `onScroll` (?). Also some duplicated key errors are displayed on the console 👇🏼 

[Screencast from 02-05-23 17:13:50.webm](https://user-images.githubusercontent.com/6906348/235713733-4bf3a070-80dd-4691-8bca-d8b51900b5c0.webm)

**Review by weberjavi — APPROVED** [review]
Thanks for the updates 🙇🏼

**Comment by j8seangel** [comment]
Closing in favour of https://github.com/GlobalFishingWatch/frontend/pull/2028
